src/areastore.h

```diff
@@ -0,0 +1,202 @@
+/*
+Minetest
+Copyright (C) 2015 est31 <mtest31@outlook.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef AREASTORE_H_
+#define AREASTORE_H_
+
+#include "irr_v3d.h"
+#include "noise.h" // for PcgRandom
+#include <map>
+#include <list>
+#include <vector>
+#include <istream>
+#include "util/container.h"
+#include "util/numeric.h"
+#ifndef ANDROID
+	#include "cmake_config.h"
+#endif
+#if USE_SPATIAL
+	#include <spatialindex/SpatialIndex.h>
+	#include "util/serialize.h"
+#endif
+
+#define AST_EXTREMIFY(min, max, pa, pb) \
+	min.X = MYMIN(pa.X, pb.X);      \
+	min.Y = MYMIN(pa.Y, pb.Y);      \
+	min.Z = MYMIN(pa.Z, pb.Z);      \
+	max.X = MYMAX(pa.X, pb.X);      \
+	max.Y = MYMAX(pa.Y, pb.Y);      \
+	max.Z = MYMAX(pa.Z, pb.Z);
+
+struct Area {
+	Area(const v3s16 &minedge, const v3s16 &maxedge)
+	{
+		data = NULL;
+		datalen = 0;
+		this->minedge = minedge;
+		this->maxedge = maxedge;
+	}
+
+	Area()
+	{
+		data = NULL;
+		datalen = 0;
+	}
+
+	~Area()
+	{
+		delete[] data;
```

Comment:
Yes, I have stumbled over this myself already. Do you have an idea how to make this better?